_posts/2018-05-02-load-balanced-smart-proxies-with-haproxy.md

```diff
@@ -0,0 +1,138 @@
+
+### Preparing `custom-hiera.yaml`
+We need to configure Pulp to serve redirects in the streamer code to the name of the load-balancer, but this option is not exposed in the installer. Thus edit `/etc/foreman-installer/custom-hiera.yaml` and add the following line at the end:
+
+```yaml
+pulp::lazy_redirect_host: smartproxy.example.test
+```
+### Running the installer
+As only the first Smart Proxy will host a PuppetCA, we configure that explicitly during the installation and also configure the Puppet certificate to include the `smartproxy.example.test` name:
+```
+--puppet-dns-alt-names smartproxy.example.test
+--puppet-ca-server proxy01.example.test
+--foreman-proxy-puppetca true
+--puppet-server-ca true
+```
+
+## Installing the other Smart Proxies
+The installation of all other Smart Proxies is almost identical to the first one. We just have to prepare the Puppet certs on the first one (as it's the only one having a PuppetCA) and disable the PuppetCA in the installer.
+
+### Generating Katello certificates
+This step is identical to the steps used to generate the certificates for the first smart proxy. Simply use `foreman-proxy-certs-generate` as described above.
+
+### Generating Puppet certificates
+On the first smart proxy, generate the Puppet certificates by running
+
+```shell
+puppet cert generate proxy02.example.test --dns_alt_names=smartproxy.example.test
+```
+
+Now copy `/etc/puppetlabs/puppet/ssl/*/proxy02.example.test.pem` and `/etc/puppetlabs/puppet/ssl/certs/ca.pem` to the second smart proxy and ensure the permissions are identical to the ones on the first host. The `puppet` user is created after installing the `puppetserver` package, so you might want to do that first. Also don't forget to check the SELinux contexts of the files you copied over (hint: `ls -lZ`) :)
+
+Repeat this step for all smart proxies that are going to sit behind the same load balancer.
+
+### Preparing `custom-hiera.yaml`
+These steps are identical to the steps used on the first Smart Proxy.
+
+### Running the installer
+As only the first Smart Proxy will host a PuppetCA, we point the Puppet agent to it explicitly during the installation and also configure the Puppet certificate to include the `smartproxy.example.test` name:
+```
+--puppet-dns-alt-names smartproxy.example.test
+--puppet-ca-server proxy01.example.test
+--foreman-proxy-puppetca false
+--puppet-server-ca false
+```
+## Installing HAProxy
+
+As HAProxy will listen on ports not "owned" by it, SELinux will be default prevent that. For the sake of this post, we'll run the whole machine in permissive mode, but it's an open TODO item to investigate how to properly confine HAProxy. Also note that if you are using a different load balancer technology, you can use this configuration as reference.
```

Comment:
The requirement to run the machine in permissive mode is IMHO worth a note in the caveats section, because it might be something for people to take into consideration before setting this up.